instance/static/js/src/instance.js

```diff
@@ -122,12 +122,33 @@ app.controller("Index", ['$scope', 'Restangular', 'OpenCraftAPI', '$q',
             } else if(message.data.type === 'instance_log') {
                 if($scope.selected.instance && $scope.selected.instance.id === message.data.instance_id) {
                     $scope.$apply(function(){
-                        $scope.selected.instance.log_text += message.data.log_entry + '\n';
+                        $scope.selected.instance.log_entries.push(message.data.log_entry);
                     });
                 }
             }
         };
 
+        $scope.error_log = function() {
+            if (!$scope.selected.instance) {
+                return;
+            }
+            var log_entries = $scope.selected.instance.log_entries;
+            var error = [];
+            var counting = false;
+            for (var i = log_entries.length; i > 0; i--) {
+                var entry = log_entries[i - 1];
+                if (entry.level != 'ERROR' && entry.level != 'CRITICAL') {
+                    if (counting) {
+                        return error.reverse();
+                    }
```

Comment:
Actually, I changed my mind here; I think it is more coherent and maybe more helpful if we display all the errors we got, since one can have contributed to the other.